api4/user.go

```diff
@@ -1670,11 +1670,16 @@ func login(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	c.LogAuditWithUserId(id, "attempt - login_id="+loginId)
 
-	user, err := c.App.AuthenticateUserForLogin(id, loginId, password, mfaToken, ldapOnly)
+	user, err := c.App.AuthenticateUserForLogin(id, loginId, password, mfaToken, cwsToken, ldapOnly)
 	if err != nil {
 		c.LogAuditWithUserId(id, "failure - login_id="+loginId)
-		c.Err = err
-		return
+		if isCWSLogin(c, cwsToken) {
+			http.Redirect(w, r, *c.App.Config().ServiceSettings.SiteURL, 302)
```

Comment:
This redirects to reload the login page if not the user gets a JSON API response with the error